akka-http/src/main/scala/akka/http/scaladsl/server/directives/ReverseProxyDirectives.scala

```diff
@@ -23,59 +23,70 @@ import scala.concurrent.Future
+    forward(authority, configOption).tapply(x ⇒ complete(x._1))
   }
 
-  def forward(authority: Authority, circuitBreakerOption: Option[CircuitBreaker] = None)(implicit system: ActorSystem): Directive1[Try[HttpResponse]] = {
+  def forward(authority: Authority, configOption: Option[ReverseProxyConfig] = None): Directive1[Try[HttpResponse]] = {
     extractUri.flatMap { uri ⇒
       val forwardUri: Uri = uri.withAuthority(authority)
-      forwardToUri(forwardUri, circuitBreakerOption)
+      forwardToUri(forwardUri, configOption)
     }
   }
 
-  def forwardUnmatchedPath(authority: Authority, circuitBreakerOption: Option[CircuitBreaker] = None)(implicit system: ActorSystem): Directive1[Try[HttpResponse]] = {
+  def forwardUnmatchedPath(authority: Authority, configOption: Option[ReverseProxyConfig] = None): Directive1[Try[HttpResponse]] = {
     extractUnmatchedPath.flatMap { unmatched ⇒
       extractUri.flatMap { uri ⇒
         val forwardUri: Uri = uri.withAuthority(authority).withPath(unmatched)
-        forwardToUri(forwardUri, circuitBreakerOption)
+        forwardToUri(forwardUri, configOption)
       }
     }
   }
 
-  def forwardToUri(uri: Uri, circuitBreakerOption: Option[CircuitBreaker] = None)(implicit system: ActorSystem): Directive1[Try[HttpResponse]] = {
-    extractMaterializer.flatMap { implicit mat ⇒
-      val requestUriInterceptor = ProxyUriRequestInterceptor(uri).andThen(ProxyHeaderRequestInterceptor)
-      //TODO: select executor from configuration
-      val executor = CachedHostConnectionPoolRequestExecutor()
-      forwardRequest(executor, requestUriInterceptor, circuitBreakerOption)
+  def forwardToUri(uri: Uri, configOption: Option[ReverseProxyConfig] = None): Directive1[Try[HttpResponse]] = {
+    val requestUriInterceptor = ProxyUriRequestInterceptor(uri).andThen(ProxyHeaderRequestInterceptor)
+    mapRequest(requestUriInterceptor).tflatMap { _ ⇒
+      forwardRequest(configOption)
     }
   }
 
-  def forwardRequest(requestExecutor: RequestExecutor, requestInterceptor: RequestInterceptor, circuitBreakerOption: Option[CircuitBreaker] = None): Directive1[Try[HttpResponse]] = {
+  def forwardRequest(configOption: Option[ReverseProxyConfig] = None): Directive1[Try[HttpResponse]] = {
     extractExecutionContext.flatMap { implicit ec ⇒
-      extractMaterializer.flatMap { implicit mat ⇒
-        extractRequest.flatMap { request ⇒
-          val interceptedRequest: HttpRequest = requestInterceptor(request)
-          val responseFuture: Future[HttpResponse] = requestExecutor(interceptedRequest)
-          circuitBreakerOption.fold(onComplete(responseFuture))(circuitBreaker ⇒ onCompleteWithBreaker(circuitBreaker)(responseFuture))
+      extractRequest.flatMap { request ⇒
+        getOrCreateDefaultConfig(configOption).flatMap { reverseProxyConfig ⇒
+          val responseFuture: Future[HttpResponse] = reverseProxyConfig.requestExecutor.apply(request)
```

Comment:
I'm unsure which executor the future will execute on here, do people have a way of configuring that?